packages/bpk-theming/src/BpkThemeProvider.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import validateThemeAttributes from './themeAttributes';
+
+class BpkThemeProvider extends React.Component {
+  constructor(props) {
+    super(props);
+    this.updateCssVariables = this.updateCssVariables.bind(this);
+  }
+
+  componentDidMount() {
+    this.updateCssVariables();
+  }
+
+  componentDidUpdate() {
+    this.updateCssVariables();
+  }
+
+  updateCssVariables() {
+    this.div.style = null;
+    const theme = validateThemeAttributes(this.props.theme);
+    Object.keys(theme).forEach((variableName) => {
+      const cssName = variableName.replace(/([A-Z])/g, variable => `-${variable.toLowerCase()}`);
+      const value = theme[variableName];
+      this.div.style.setProperty(`--${cssName}`, value);
```

Comment:
I've justed started thingking about the affect of this code and any risks of repaints/layout thrashing as a consequence. Maybe using the React `style` object would be better as I assume React batches it in a single DOM manipulation step